alpha

pbearne opened the pull request:
we have code
I have no idea if the imagick code path works

## Summary

<!-- Please reference the issue this PR fixes. If this PR does not fix the entire issue, change this to Addresses #... instead. -->
Fixes #

## Relevant technical choices

<!-- Please describe your changes. -->

## Checklist

- [x] PR has either `[Focus]` or `Infrastructure` label.
- [x] PR has a `[Type]` label.
- [ ] PR has a milestone or the `no milestone` label.

pbearne committed:
alpha

we have code
I have no idea if the imagick code path works
not sure the js to remove bg works

pbearne committed:
alpha

we have code
I have no idea if the imagick code path works
not sure the js to remove bg works

pbearne committed:
added some tests

pbearne committed:
added some tests

mehigh commented on modules/images/dominant-color/load.php:
```diff
@@ -0,0 +1,425 @@
+
+	public function add_scripts() {
+		?>
+<script>
```
@pbearne there is no need for any JavaScript, so please remove the inline one here. With the images being opaque, the presence, or lack of a background color makes no difference.

mehigh commented on modules/images/dominant-color/load.php:
```diff
@@ -0,0 +1,425 @@
+			$extra_class = '';
+
+			if ( ! isset( $image_meta['has_transparency'] ) ) {
+				$data        .= ' data-hasTransparency="false"';
```
please use data-has-transparency= as an attribute name instead.

mehigh commented on modules/images/dominant-color/load.php:
```diff
@@ -0,0 +1,425 @@
+			if ( ! isset( $image_meta['has_transparency'] ) ) {
+				$data        .= ' data-hasTransparency="false"';
+			} else {
+				$data .= ' data-hasTransparency="true"';
```
Here too, don't use the camel-case in the HTML attributes.
Here's the why
https://usefulangle.com/post/172/javascript-access-data-attributes

`data-post-id=`

is accessed in JS as
`.dataset.postId`

pbearne replied:
done

mehigh commented on modules/images/dominant-color/load.php:
```diff
@@ -0,0 +1,425 @@
+		<style>
+			img[data-dominantcolor]:not(.has-transparency) {
+				background-color: var(--dominant-color);
+				background-clip: content-box, padding-box;
```
Use only 
`background-clip: content-box;`
https://developer.mozilla.org/en-US/docs/Web/CSS/background-clip

pbearne replied:
done

mehigh commented on modules/images/dominant-color/load.php:
```diff
@@ -0,0 +1,425 @@
+			$context = current_filter();
+		}
+
+		if ( ! preg_match_all( '/<(img|iframe)\s[^>]+>/', $content, $matches, PREG_SET_ORDER ) ) {
```
Why is this searching for iframes?
It should probably be just img tags.

pbearne replied:
this is a direct copy from  core 
I can remove this or keep so we can keep parity with core

mehigh replied:
I would at the very least remove the |iframe from it. we need to bail as soon as we don't have an image to work with.

spacedmonkey replied:
Let's remove iframe here. Not needed.

mehigh commented on modules/images/dominant-color/load.php:
```diff
@@ -0,0 +1,425 @@
+		// List of the unique `img` tags found in $content.
+		$images = array();
+
+		// List of the unique `iframe` tags found in $content.
```
This probably needs dismissing (probably part of a copy paste)

pbearne replied:
removed

mehigh commented on modules/images/dominant-color/load.php:
```diff
@@ -0,0 +1,425 @@
+	}
+
+	/**
+	 * Add checkbox setting to enable/disable blurhash for a given media.
```
update comment to reference dominant color instead of blurhash.

pbearne replied:
done

mehigh commented on modules/images/dominant-color/load.php:
```diff
@@ -0,0 +1,425 @@
+	 */
+	public function add_dominant_color_media_setting( array $form_fields, WP_Post $post ) {
+		$image_meta     = wp_get_attachment_metadata( $post->ID );
+		$checked_text   = isset( $image_meta['show_dominant_color'] ) ? 'checked' : '';
```
Why are we looking on an image by image basis whether the dominant color would be shown?
We should probably be doing this on a global level (does the theme support this feature or not, and if it's supported, plug it in everywhere)

pbearne replied:
We have the ability to disable in the media settings

mehigh commented on modules/images/dominant-color/load.php:
```diff
@@ -0,0 +1,425 @@
+			if ( ! isset( $image_meta['show_dominant_color'] ) ) {
+				// If enabling blurhash from media setting and image meta doesn't have any, generate a new one.
+				$image_meta                        = $this->dominant_color_metadata( $image_meta, $attachment_id );
+				$image_meta['show_dominant_color'] = $attachment['show_dominant_color'] ? '1' : '0';
```
i might've been wrong in an above comment, sorry.. this flag only refers to it being computed?

mehigh commented on modules/images/dominant-color/load.php:
```diff
@@ -0,0 +1,425 @@
+				wp_update_attachment_metadata( $attachment_id, $image_meta );
+			}
+		} elseif ( isset( $image_meta['dominant_color'] ) ) {
+			// If disabling blurhash from media setting and blurhash set in image meta, unset it.
```
another blurhash reference

pbearne replied:
done

mehigh commented on modules/images/dominant-color/load.php:
```diff
@@ -0,0 +1,425 @@
+	 */
+	private function get_dominant_color_GD( $file, $default_color = 'eee' ) {
+
+		wp_raise_memory_limit( 'image' );
```
Is this necessary outside of debugging?

pbearne replied:
This code is in core in the media functions
I don't know if it is needed or not

mehigh commented on modules/images/dominant-color/load.php:
```diff
@@ -0,0 +1,425 @@
+	 * @return bool
+	 */
+	function get_has_transparency_GD( $file ) {
+		wp_raise_memory_limit( 'image' );
```
same here. what's the usecase that requires it?

mehigh reviewed (changes requested):
Thanks for the awesome work here, it's starting to come together nicely. I've sent in a couple of request.

Commit:
Added some more tests

Commit:
Added some more tests

Commit:
code review changes

Commit:
code review changes

spacedmonkey commented on modules/images/dominant-color/load.php:
```diff
@@ -0,0 +1,425 @@
+			add_filter( 'wp_dominant_color_img_tag_add_adjust', [ $this, 'tag_add_adjust' ], 10, 3 );
+		}
+
+		add_filter( 'attachment_fields_to_edit', [ $this, 'add_dominant_color_media_setting' ], 10, 2 );
+		add_filter( 'attachment_fields_to_save', [ $this, 'save_dominant_color_media_setting' ], 10, 2 );
```
I don't think we need the ability to save colour here. Let's generate this in the background for now and not at UI to edit it.

spacedmonkey commented on modules/images/dominant-color/load.php:
```diff
@@ -0,0 +1,425 @@
+		add_filter( 'wp_generate_attachment_metadata', [ $this, 'has_transparency_metadata' ], 10, 2 );
+
+		// do we have the new filter or are duplicating core the functions?
+		if ( has_filter( 'wp_img_tag_add_adjust' ) ) {
```
Has_filter does not work this way. It check if something has hooked into that filter. Let's remove this until that PR is merged into core.

spacedmonkey commented on modules/images/dominant-color/load.php:
```diff
@@ -0,0 +1,425 @@
+	 * @return bool
+	 */
+	public function get_has_transparency( $id ) {
+		$file         = get_attached_file( $id );
```
We need to check that the upload is an image here.

spacedmonkey commented on modules/images/dominant-color/load.php:
```diff
@@ -0,0 +1,425 @@
+	 *
+	 * @return array $metadata
+	 */
+	public function has_transparency_metadata( $metadata, $attachment_id ) {
```
What happens for none images?

spacedmonkey commented on modules/images/dominant-color/load.php:
```diff
@@ -0,0 +1,425 @@
+		$has_transparency = $this->get_has_transparency( $attachment_id );
+
+		if ( ! empty( $has_transparency ) ) {
+			$metadata['has_transparency'] = $has_transparency;
```
Let's always save this value, either true or false. But only for images.

spacedmonkey commented on modules/images/dominant-color/load.php:
```diff
@@ -0,0 +1,425 @@
+	/**
+	 * @return void
+	 */
+	public function add_styles() {
```
Use [wp_add_inline_style](https://developer.wordpress.org/reference/functions/wp_add_inline_style/) instead here.

spacedmonkey replied:
^

spacedmonkey commented on modules/images/dominant-color/load.php:
```diff
@@ -0,0 +1,425 @@
+		<?php
+	}
+
+	public function add_scripts() {
```
Let's use [wp_add_inline_script](https://developer.wordpress.org/reference/functions/wp_add_inline_script/) here.

pbearne replied:
removed JS

spacedmonkey commented on modules/images/dominant-color/load.php:
```diff
@@ -0,0 +1,425 @@
+	public function add_scripts() {
+		?>
+<script>
+	(function (window, document) {
```
```suggestion
	(function () {
```

spacedmonkey replied:
I don't think these are needed.

spacedmonkey commented on modules/images/dominant-color/load.php:
```diff
@@ -0,0 +1,425 @@
+			var image = event.target;
+			image.style.setProperty('--dominant-color', '');
+			});
+	})(window, document);
```
```suggestion
	})();
```

Same here.

pbearne replied:
removed JS

spacedmonkey commented on modules/images/dominant-color/load.php:
```diff
@@ -0,0 +1,425 @@
+		?>
+<script>
+	(function (window, document) {
+		'use strict';
```
```suggestion
```

pbearne replied:
removed JS

spacedmonkey commented on modules/images/dominant-color/load.php:
```diff
@@ -0,0 +1,425 @@
+
+
+
+	/**
+	 * @return void
+	 */
+//	public function enqueue_plugin_scripts() {
+//		wp_enqueue_script( 'blurhash', plugin_dir_url( __FILE__ ) . 'dist/blurhash.js' );
+//	}
```
```suggestion
```

pbearne replied:
removed JS

spacedmonkey reviewed (changes requested):
Please ensure that all methods, have PHP docs, method visibility. All ensure than all php lints pass.

spacedmonkey commented on modules/images/dominant-color/load.php:
```diff
@@ -0,0 +1,425 @@
+		if ( has_filter( 'wp_img_tag_add_adjust' ) ) {
+			add_filter( 'wp_img_tag_add_adjust', [ $this, 'tag_add_adjust' ], 20, 3 );
+		} else {
+			add_filter( 'the_content', [ $this, 'filter_content_tags' ], 20 );
```
This needs to be applied to all filter that wp_filter_content_tags is applied too 

So `the_excerpt`, `widget_text_content` and `widget_block_content`

spacedmonkey commented on modules/images/dominant-color/load.php:
```diff
@@ -0,0 +1,425 @@
+			add_filter( 'the_content', [ $this, 'filter_content_tags' ], 20 );
+			add_filter( 'wp_dominant_color_img_tag_add_adjust', [ $this, 'tag_add_adjust' ], 10, 3 );
+		}
+
```
I would also apply to `wp_get_attachment_image_attributes` filter.

spacedmonkey commented on tests/modules/images/domiant-color/get-dominant-color-GD-test.php:
```diff
@@ -0,0 +1,48 @@
+ * Tests for webp-uploads module.
+ *
+ * @package performance-lab
+ * @group   webp-uploads
```
Change group.

spacedmonkey commented:
To keep the implementation a lot cleaner, let's do the following. 

Create a new class called `WP_Image_Editor_GD_With_Color` that extends `WP_Image_Editor_GD`.
Move all logic for getting color and transparent to this new class. 
Filter `wp_image_editors` and remove `WP_Image_Editor_GD` and replace with  `WP_Image_Editor_GD_With_Color`

Repeat for Imagek.

spacedmonkey commented on tests/modules/images/domiant-color/get-dominant-color-GD-test.php:
```diff
@@ -0,0 +1,48 @@
+ * @group   webp-uploads
+ */
+
+class get_dominant_color_GD_Tests extends WP_UnitTestCase {
```
Once the GD / imagek classes exist, these classes should be renamed and implement covers notes.

spacedmonkey replied:
Let's name these test classes, similar to the class.

spacedmonkey commented on tests/modules/images/domiant-color/get-dominant-color-GD-test.php:
```diff
@@ -0,0 +1,48 @@
+	function provider_get_dominant_color() {
+		return array(
+
+			'jpg' => array(
```
Let's tests a PNG, a gif and an animated gif.

spacedmonkey replied:
I would also test an image an image that is just a transparent pixels.

spacedmonkey commented on tests/modules/images/domiant-color/get-dominant-color-Imagick-test.php:
```diff
@@ -0,0 +1,48 @@
+	function set_up() {
+		parent::set_up(); // TODO: Change the autogenerated stub
+
+		add_filter( 'wp_image_editors', static function (){ return array('WP_Image_Editor_Imagick'); } );
```
Add checks for none image files to ensure that this code does not generate.

spacedmonkey commented on tests/modules/images/domiant-color/get-dominant-color-GD-test.php:
```diff
@@ -0,0 +1,48 @@
+	function set_up() {
+		parent::set_up(); // TODO: Change the autogenerated stub
+
+		add_filter( 'wp_image_editors', static function (){ return array('WP_Image_Editor_GD'); } );
```
Add checks for none image files to ensure that this code does not generate.

spacedmonkey commented on modules/images/dominant-color/load.php:
```diff
@@ -0,0 +1,425 @@
+	 *
+	 * @return string
+	 */
+	private function get_dominant_color_Imagick( $file, $default_color = 'eee' ) {
```
I am not sure about the use of default color, this might have side effects we do not understand yet.

spacedmonkey commented on modules/images/dominant-color/load.php:
```diff
@@ -0,0 +1,425 @@
+		wp_raise_memory_limit( 'image' );
+		try {
+			// phpcs:ignore WordPress.PHP.NoSilencedErrors.Discouraged
+			if ( $original_image = new Imagick( $file ) ) {
```
When we extend the `WP_Image_Editor_Imagick` we will not needed.

spacedmonkey commented on modules/images/dominant-color/load.php:
```diff
@@ -0,0 +1,425 @@
+	 */
+	function get_has_transparency_GD( $file ) {
+		wp_raise_memory_limit( 'image' );
+		if ( $original_image = @imagecreatefromstring( @file_get_contents( $file ) ) ) {
```
Coding standards do not allow for this.

spacedmonkey commented on modules/images/dominant-color/load.php:
```diff
@@ -0,0 +1,425 @@
+		if ( $original_image = @imagecreatefromstring( @file_get_contents( $file ) ) ) {
+			return imageistruecolor( $original_image ) ? imagecolortransparent( $original_image ) > 0 : imagecolorsforindex( $original_image, imagecolortransparent( $original_image ) )['alpha'] > 0;
+		} else {
+			return false;
```
Errors like this, should return a `WP_Error` object.

spacedmonkey commented on modules/images/dominant-color/load.php:
```diff
@@ -0,0 +1,425 @@
+				// phpcs:ignore WordPress.PHP.NoSilencedErrors.Discouraged
+				return (bool) @$original_image->getImageAlphaChannel() == Imagick::ALPHACHANNEL_TRANSPARENT;
+			} catch ( Exception $e ) {
+				return false;
```
Errors should return WP_errors.

spacedmonkey commented on modules/images/dominant-color/load.php:
```diff
@@ -0,0 +1,425 @@
+				return false;
+			}
+		} else {
+			return false;
```
https://github.com/xwp/performance/pull/3/files#r832170030

spacedmonkey commented on modules/images/dominant-color/load.php:
```diff
@@ -0,0 +1,425 @@
+
+				return dechex( $color['r'] ) . dechex( $color['g'] ) . dechex( $color['b'] );
+			} else {
+				return $default_color;
```
Let's not use a default color here.

spacedmonkey commented on modules/images/dominant-color/load.php:
```diff
@@ -0,0 +1,425 @@
+				return $default_color;
+			}
+		} catch ( Exception $e ) {
+			return $default_color;
```
Return a wp error here.

spacedmonkey commented:
This is a good start, let's work on the feedback and I will re-review.

pbearne committed:
removed iframe

pbearne committed:
removed iframe

Commit:
code review changes
created classes for image functions
Updated tests

NOT all tests are passsing

Commit:
code review changes
created classes for image functions
Updated tests

NOT all tests are passsing

Commit:
change filter to match core change

Commit:
change filter to match core change

spacedmonkey commented on modules/images/dominant-color/load.php:
```diff
@@ -0,0 +1,407 @@
+		$has_transparency = false;
+		if ( isset( $image_meta['has_transparency'] ) ) {
+			$has_transparency = $image_meta['has_transparency'];
+		}
```
```suggestion
		$has_transparency = isset( $image_meta['has_transparency'] ) ? $image_meta['has_transparency'] : false;
```

spacedmonkey commented on modules/images/dominant-color/load.php:
```diff
@@ -0,0 +1,407 @@
+	 */
+	public function tag_add_adjust_to_image_attributes( $attr, $attachment ) {
+
+		$image_meta = wp_get_attachment_metadata( $attachment->ID );
```
I would check to see if this return an array.

spacedmonkey commented on modules/images/dominant-color/load.php:
```diff
@@ -0,0 +1,407 @@
+		return $content;
+	}
+
+	public function set_wp_image_editors() {
```
Need a php docs.

spacedmonkey commented on modules/images/dominant-color/load.php:
```diff
@@ -0,0 +1,407 @@
+
+		add_filter( 'wp_generate_attachment_metadata', array( $this, 'dominant_color_metadata' ), 10, 2 );
+		add_filter( 'wp_generate_attachment_metadata', array( $this, 'has_transparency_metadata' ), 10, 2 );
+
+		// do we have the new filter or are duplicating core the functions?
+		if ( has_filter( 'wp_content_img_tag' ) ) {
+			add_filter( 'wp_content_img_tag', array( $this, 'tag_add_adjust' ), 20, 3 );
+		} else {
+			add_filter( 'the_content', array( $this, 'filter_content_tags' ), 20 );
+			add_filter( 'the_excerpt', array( $this, 'filter_content_tags' ), 20 );
+			add_filter( 'widget_text_content ', array( $this, 'filter_content_tags' ), 20 );
+			add_filter( 'widget_block_content', array( $this, 'filter_content_tags' ), 20 );
+			add_filter( 'wp_dominant_color_img_tag_add_adjust', array( $this, 'tag_add_adjust' ), 10, 3 );
+		}
```
```suggestion
	add_filter( 'wp_content_img_tag', array( $this, 'tag_add_adjust' ), 20, 3 );
	add_filter( 'wp_dominant_color_img_tag_add_adjust', array( $this, 'tag_add_adjust' ), 10, 3 );
	$filters = array( 'the_content', 'the_excerpt', 'widget_text_content', 'widget_block_content' );
	foreach( $filters as $filter ) {
	       add_filter( $filter, array( $this, 'filter_content_tags' ), 20 );
	}	
```

Let's do this, it's cleaner.

spacedmonkey commented on modules/images/dominant-color/load.php:
```diff
@@ -0,0 +1,407 @@
+	 * @return array $metadata
+	 */
+	public function dominant_color_metadata( $metadata, $attachment_id ) {
+
```
```suggestion
          if( !wp_attachment_is_image( $attachment_id ) ) {
               return $metadata;
          }
```

spacedmonkey commented on modules/images/dominant-color/load.php:
```diff
@@ -0,0 +1,407 @@
+	 *
+	 * @return array $metadata
+	 */
+	public function has_transparency_metadata( $metadata, $attachment_id ) {
```
```suggestion
	public function has_transparency_metadata( $metadata, $attachment_id ) {

          if( !wp_attachment_is_image( $attachment_id ) ) {
               return $metadata;
          }
```
```

spacedmonkey commented on modules/images/dominant-color/load.php:
```diff
@@ -0,0 +1,407 @@
+	 *
+	 * @return array|false Return associative array if success, return `false` for otherwise.
+	 */
+	private function webpinfo( $file ) {
```
How does this with wp_get_webp_info in core? Can we use that?

pbearne replied:
this is just looking for alpha channel
renamed to make clear

spacedmonkey commented:
Ensure that all unit tests and php lint pass. That I can re-review.

pbearne committed:
Merge branch 'WordPress:trunk' into INITS-199_domiant_color

pbearne committed:
Update modules/images/dominant-color/load.php

Co-authored-by: Jonny Harris <spacedmonkey@users.noreply.github.com>

pbearne committed:
Update modules/images/dominant-color/load.php

Co-authored-by: Jonny Harris <spacedmonkey@users.noreply.github.com>

pbearne committed:
Update modules/images/dominant-color/load.php

Co-authored-by: Jonny Harris <spacedmonkey@users.noreply.github.com>

pbearne committed:
Update modules/images/dominant-color/load.php

Co-authored-by: Jonny Harris <spacedmonkey@users.noreply.github.com>

pbearne committed:
Update modules/images/dominant-color/load.php

Co-authored-by: Jonny Harris <spacedmonkey@users.noreply.github.com>

pbearne committed:
Update modules/images/dominant-color/load.php

Co-authored-by: Jonny Harris <spacedmonkey@users.noreply.github.com>

pbearne committed:
Update modules/images/dominant-color/load.php

Co-authored-by: Jonny Harris <spacedmonkey@users.noreply.github.com>

pbearne committed:
Update modules/images/dominant-color/load.php

Co-authored-by: Jonny Harris <spacedmonkey@users.noreply.github.com>

pbearne committed:
Merge branch 'WordPress:trunk' into INITS-199_domiant_color

pbearne committed:
Added test files

got get_has_transparency() working

pbearne committed:
Added test files

got get_has_transparency() working

pbearne committed:
comments

pbearne committed:
comments

pbearne committed:
fixed test

pbearne committed:
fixed test

pbearne committed:
fix linting errors

pbearne committed:
fix linting errors

pbearne committed:
more linting

pbearne committed:
more linting

pbearne committed:
the last linting

pbearne committed:
the last linting

pbearne committed:
removed old conditional code

pbearne committed:
removed old conditional code

pbearne committed:
Merge branch 'INITS-199_domiant_color' of github.com:xwp/performance into INITS-199_domiant_color

pbearne committed:
refactored to get the linting to work

pbearne committed:
refactored to get the linting to work

pbearne committed:
added filer to allow fine grain control
enable_dominant_color_for_image

pbearne committed:
added filer to allow fine grain control
enable_dominant_color_for_image

pbearne committed:
more linting fixes

pbearne committed:
more linting fixes

manuelRod commented:
@pbearne shouldn't we add more test coverage? we would need to cover functions like tag_add_adjust_to_image_attributes, tag_add_adjust, etc, to make sure all regular expressions and string replacements work as expected.

manuelRod committed:
lint fix and directory wording typo

manuelRod committed:
lint fix and directory wording typo

manuelRod committed:
adding some extra testing

manuelRod committed:
adding some extra testing

spacedmonkey committed:
Fix lints

spacedmonkey committed:
Fix lints

spacedmonkey commented on phpinfo.php:
```diff
@@ -0,0 +1,2 @@
+<?php
```
Let's remove this line.

spacedmonkey commented on modules/images/dominant-color/load.php:
```diff
@@ -0,0 +1,15 @@
+/**
+ * Module class.
+ */
+include 'class-wp-dominant-color.php';
```
Require once

spacedmonkey commented on tests/modules/images/webp-uploads/webp-uploads-test.php:
```diff
@@ -3,7 +3,7 @@
  *
  * @package performance-lab
- * @group   webp-uploads
+ * @group   performance-lab
```
Why is this being changed?

spacedmonkey replied:
```suggestion
 * @group   webp-uploads
```

spacedmonkey commented on modules/images/dominant-color/class-wp-image-editor-gd-with-color.php:
```diff
@@ -0,0 +1,67 @@
+ * WordPress Image Editor Class for Image Manipulation through GD
+ * with dominant color detection
+ *
+ * @since 6.0.0
```
This will not going WordPress 6.0.0. I would either make it performance plugin since, so 1.1.0 or 6.x for WP

spacedmonkey commented on composer.lock:
```diff
@@ -157,27 +157,27 @@
     "packages-dev": [
```
Are the changes in this lock file needed for this change? If not, this change should be reverted.

spacedmonkey commented on tests/modules/images/dominant-color/dominant-color-test.php:
```diff
@@ -0,0 +1,115 @@
+	 */
+	public function test_dominant_color_metadata( $image_path, $expected_color, $expected_transparency, $color_is_light ) {
+		// Non existing attachment.
+		$dominant_color_metadata = ( new WP_Dominant_Color() )->dominant_color_metadata( array(), 1 );
```
There doesn't need to be multiple new instances of the same class in a test. You can create one instance and then reuse it.

pbearne committed:
Delete phpinfo.php

pbearne committed:
Delete phpinfo.php

pbearne committed:
code review feedback fixes

pbearne committed:
code review feedback fixes

pbearne committed:
code review feedback fixes

pbearne committed:
code review feedback fixes

pbearne committed:
reverted lock

pbearne committed:
Merge remote-tracking branch 'origin/INITS-199_domiant_color' into INITS-199_domiant_color

# Conflicts:
#	composer.lock

spacedmonkey committed:
Use inline styles

spacedmonkey committed:
Merge branch 'INITS-199_domiant_color' of github.com:xwp/performance into INITS-199_domiant_color

spacedmonkey committed:
Revert lock file

spacedmonkey committed:
Apply suggestions from code review

spacedmonkey committed:
Small tweaks

spacedmonkey committed:
Merge branch 'INITS-199_domiant_color' of github.com:xwp/performance into INITS-199_domiant_color

spacedmonkey committed:
Fix lint

spacedmonkey reviewed (approved):
Made some final tweaks to PR to tidy up. This is ready for submitting to WP.

pbearne committed:
Merge remote-tracking branch 'origin/INITS-199_domiant_color' into INITS-199_domiant_color

# Conflicts:
#	composer.lock

pbearne committed:
added code to get and save dominant color if not already set in image meta

pbearne committed:
whitespaces

pbearne committed:
whitespaces

pbearne committed:
Merge branch 'WordPress:trunk' into INITS-199_domiant_color

pbearne committed:
Removed class from code

pbearne committed:
Merge remote-tracking branch 'origin/INITS-199_domiant_color' into INITS-199_domiant_color

pbearne committed:
Removed class file

pbearne committed:
fixed tests

pbearne committed:
fixed $this ref

pbearne committed:
Stopped adding the filters not needed for WP version 6 onwards as we have a new filter (wp_content_img_tag) to use

pbearne committed:
White space tidy

pbearne committed:
Changed how we back-fill the previously uploaded images.
We now by default create run once cron tasks, so we don't slow the page load down

pbearne committed:
whitespace

pbearne committed:
Update modules/images/dominant-color/class-wp-image-editor-gd-with-color.php

Co-authored-by: Felix Arntz <felixarntz@users.noreply.github.com>

pbearne committed:
Update modules/images/dominant-color/class-wp-image-editor-gd-with-color.php

Co-authored-by: Felix Arntz <felixarntz@users.noreply.github.com>

pbearne committed:
Update modules/images/dominant-color/class-wp-image-editor-gd-with-color.php

Co-authored-by: Felix Arntz <felixarntz@users.noreply.github.com>

pbearne committed:
Update modules/images/dominant-color/class-wp-image-editor-imagick-with-color.php

Co-authored-by: Felix Arntz <felixarntz@users.noreply.github.com>

pbearne committed:
Update modules/images/dominant-color/load.php

Co-authored-by: Felix Arntz <felixarntz@users.noreply.github.com>

pbearne committed:
Update modules/images/dominant-color/load.php

Co-authored-by: Felix Arntz <felixarntz@users.noreply.github.com>

pbearne committed:
removed the wp_ prefix
and few other changes from review

pbearne committed:
removed the backfill from this PR and

pbearne committed:
whitespace and test file rename

pbearne committed:
whitespace and test file rename

pbearne committed:
whitespace and test file rename

pbearne committed:
Fixed the tests

pbearne committed:
Fixed the version compare
changed code to always set has_transparency in meta so early bail works

pbearne committed:
more fixes for has_transparency changes

pbearne committed:
Use Yoda Condition checks, you must.

pbearne committed:
Added test for dominant_color_color_is_light and add code to handle short hex colors

pbearne committed:
added return early to the images class function if no image object set

pbearne committed:
added return early to the images class function if no image object set

pbearne committed:
Update modules/images/dominant-color/load.php

Co-authored-by: Eugene Manuilov <eugene.manuilov@gmail.com>

pbearne committed:
Update modules/images/dominant-color/load.php

Co-authored-by: Eugene Manuilov <eugene.manuilov@gmail.com>

pbearne committed:
Update modules/images/dominant-color/load.php

Co-authored-by: Eugene Manuilov <eugene.manuilov@gmail.com>

pbearne committed:
removed the second call to fetch editor in dominant_color_get_has_transparency()

pbearne committed:
Merge remote-tracking branch 'origin/INITS-199_domiant_color' into INITS-199_domiant_color

# Conflicts:
#	modules/images/dominant-color/load.php

pbearne committed:
linting white spaces

pbearne committed:
added checks for methods before calling just in case ours class are overloaded

pbearne committed:
added checks for methods before calling just in case ours class are overloaded

pbearne committed:
added checks for methods before calling just in case ours class are overloaded

pbearne committed:
Merge remote-tracking branch 'origin/INITS-199_domiant_color' into INITS-199_domiant_color

# Conflicts:
#	modules/images/dominant-color/load.php

pbearne committed:
changed method_exists to look at the class not its name for check

pbearne committed:
changed method_exists to look at the class not its name for check

pbearne committed:
white

pbearne committed:
whitespace

pbearne committed:
whitespace

pbearne committed:
Update modules/images/dominant-color/load.php

Co-authored-by: Eugene Manuilov <eugene.manuilov@gmail.com>

pbearne committed:
Update modules/images/dominant-color/load.php

Co-authored-by: Eugene Manuilov <eugene.manuilov@gmail.com>

pbearne committed:
added filter to override Default color

pbearne committed:
Merge remote-tracking branch 'origin/INITS-199_domiant_color' into INITS-199_domiant_color

pbearne committed:
Update modules/images/dominant-color/load.php

change the code to append styles

Co-authored-by: Crisoforo Gaspar Hernández <hello@crisoforo.com>

pbearne committed:
Update modules/images/dominant-color/load.php

refactor class adding

Co-authored-by: Crisoforo Gaspar Hernández <hello@crisoforo.com>

pbearne committed:
Update modules/images/dominant-color/load.php

remove extra line

Co-authored-by: Crisoforo Gaspar Hernández <hello@crisoforo.com>

pbearne committed:
Update modules/images/dominant-color/load.php

remove extra line

Co-authored-by: Crisoforo Gaspar Hernández <hello@crisoforo.com>

pbearne committed:
whitespace fixes

pbearne committed:
whitespace fixes

pbearne committed:
Fixed test after changing default color

pbearne committed:
Update modules/images/dominant-color/load.php

Co-authored-by: Felix Arntz <felixarntz@users.noreply.github.com>

pbearne committed:
Update modules/images/dominant-color/load.php

Co-authored-by: Felix Arntz <felixarntz@users.noreply.github.com>

pbearne committed:
Changed code to not store color if we get an error back from create functions
Moved the default color to the output and not store in meta

moved @since in docs blocks

pbearne committed:
Merge remote-tracking branch 'origin/INITS-199_domiant_color' into INITS-199_domiant_color

pbearne committed:
whitespace

pbearne committed:
combined dominant_color_has_transparency_metadata into dominant_color_metadata to save duplicate code

pbearne committed:
code changes from code reviews

pbearne committed:
added filter to allow adjustment of the image meta (and to run backfill)

pbearne committed:
added $attachment_id to backfill filter

pbearne committed:
Update modules/images/dominant-color/load.php

Co-authored-by: Crisoforo Gaspar Hernández <hello@crisoforo.com>

pbearne committed:
Update modules/images/dominant-color/load.php

Co-authored-by: Crisoforo Gaspar Hernández <hello@crisoforo.com>

pbearne committed:
Update modules/images/dominant-color/load.php

Co-authored-by: Crisoforo Gaspar Hernández <hello@crisoforo.com>

pbearne committed:
Update modules/images/dominant-color/load.php

Co-authored-by: Crisoforo Gaspar Hernández <hello@crisoforo.com>

pbearne committed:
Update modules/images/dominant-color/load.php

Co-authored-by: Crisoforo Gaspar Hernández <hello@crisoforo.com>

pbearne committed:
Update modules/images/dominant-color/load.php

Co-authored-by: Crisoforo Gaspar Hernández <hello@crisoforo.com>

pbearne committed:
Update modules/images/dominant-color/load.php

Co-authored-by: Crisoforo Gaspar Hernández <hello@crisoforo.com>

pbearne committed:
Update modules/images/dominant-color/load.php

Co-authored-by: Crisoforo Gaspar Hernández <hello@crisoforo.com>

pbearne committed:
Whitespace fixes after Git updates

pbearne committed:
Whitespace fixes after Git updates

pbearne committed:
set the default $has_transparency to true

pbearne committed:
renamed function dominant_color_tag_add_adjust_to_image_attributes to dominant_color_update_attachment_image_attributes

pbearne committed:
rename editor functions
and added return error for has_transparency functions

pbearne committed:
Fix error after function rename

pbearne committed:
Merge branch 'trunk' into INITS-199_domiant_color

pbearne committed:
Update modules/images/dominant-color/load.php

Co-authored-by: Felix Arntz <felixarntz@users.noreply.github.com>

pbearne committed:
Update modules/images/dominant-color/class-dominant-color-image-editor-imagick.php

Co-authored-by: Felix Arntz <felixarntz@users.noreply.github.com>

pbearne committed:
Update modules/images/dominant-color/class-dominant-color-image-editor-imagick.php

Co-authored-by: Felix Arntz <felixarntz@users.noreply.github.com>

pbearne committed:
Merge remote-tracking branch 'origin/INITS-199_domiant_color' into INITS-199_domiant_color

pbearne committed:
added dominant color to CODEOWNERS

pbearne committed:
Add translators: notes
removed filter on meta

pbearne committed:
updated translators: notes

pbearne committed:
Fixes to test for changes after trans

pbearne committed:
rename get_dominant_color() to dominant_color_get_dominant_color()

pbearne committed:
Update modules/images/dominant-color/load.php

Co-authored-by: Felix Arntz <felixarntz@users.noreply.github.com>

pbearne committed:
removed not needed spaces

pbearne committed:
Update modules/images/dominant-color/load.php

Co-authored-by: Felix Arntz <felixarntz@users.noreply.github.com>

pbearne committed:
Update modules/images/dominant-color/load.php

Co-authored-by: Felix Arntz <felixarntz@users.noreply.github.com>

pbearne committed:
rewritten the $extra_class code to use an array and tidied the logic when adding

pbearne committed:
Merge remote-tracking branch 'origin/INITS-199_domiant_color' into INITS-199_domiant_color

pbearne committed:
rewritten the $extra_class code to use an array and tidied the logic when adding in the dominant_color_img_tag_add_dominant_color funtion

pbearne committed:
remove all 5 extra lines before returns

pbearne committed:
whitespace

pbearne committed:
fixed test

pbearne committed:
Merge branch 'WordPress:trunk' into INITS-199_domiant_color

pbearne committed:
Update modules/images/dominant-color/load.php

Co-authored-by: Jonny Harris <spacedmonkey@users.noreply.github.com>

pbearne committed:
Update modules/images/dominant-color/load.php

Co-authored-by: Jonny Harris <spacedmonkey@users.noreply.github.com>

pbearne committed:
Merge branch 'WordPress:trunk' into INITS-199_domiant_color

pbearne committed:
Update modules/images/dominant-color/load.php

Co-authored-by: Jonny Harris <spacedmonkey@users.noreply.github.com>

pbearne committed:
Update modules/images/dominant-color/load.php

Co-authored-by: Jonny Harris <spacedmonkey@users.noreply.github.com>

pbearne committed:
Update modules/images/dominant-color/load.php

Co-authored-by: Jonny Harris <spacedmonkey@users.noreply.github.com>

pbearne committed:
Removed is light code

pbearne committed:
Merge remote-tracking branch 'origin/INITS-199_domiant_color' into INITS-199_domiant_color

pbearne committed:
Removed default color filter

pbearne committed:
Removed line before returns

pbearne committed:
whitespace

pbearne committed:
whitespace

pbearne committed:
removed is light test

pbearne committed:
removed is light test

pbearne committed:
Update modules/images/dominant-color/load.php

Co-authored-by: Jonny Harris <spacedmonkey@users.noreply.github.com>

pbearne committed:
swapped the if array checks for isset checks

pbearne committed:
Merge branch 'WordPress:trunk' into INITS-199_domiant_color

pbearne committed:
reverted the if array checks removal

pbearne committed:
Update modules/images/dominant-color/class-dominant-color-image-editor-imagick.php

Co-authored-by: Eugene Manuilov <eugene.manuilov@gmail.com>

pbearne committed:
Update modules/images/dominant-color/load.php

Co-authored-by: Jonny Harris <spacedmonkey@users.noreply.github.com>

spacedmonkey committed:
Improve logic and tests.

spacedmonkey committed:
Fix lints.

spacedmonkey committed:
Merge branch 'INITS-199_domiant_color' into fix/tweaks-color

spacedmonkey committed:
More tweaks.

spacedmonkey committed:
Merge pull request #4 from xwp/fix/tweaks-color

Feedback on dominant color

pbearne committed:
Update modules/images/dominant-color/load.php

Co-authored-by: Felix Arntz <felixarntz@users.noreply.github.com>

spacedmonkey committed:
Change method names.

spacedmonkey committed:
Change data.

spacedmonkey committed:
Use __dir__.

spacedmonkey committed:
Generate color from thumbnail

spacedmonkey committed:
Force generation of images sizes.

spacedmonkey committed:
Force generate thumanils

spacedmonkey committed:
Fix lint

spacedmonkey committed:
Merge branch 'INITS-199_domiant_color' of github.com:xwp/performance into INITS-199_domiant_color

spacedmonkey committed:
Fix tests

pbearne committed:
Update modules/images/dominant-color/load.php

Co-authored-by: Felix Arntz <felixarntz@users.noreply.github.com>

pbearne committed:
Update modules/images/dominant-color/load.php

Co-authored-by: Felix Arntz <felixarntz@users.noreply.github.com>

pbearne committed:
Update modules/images/dominant-color/load.php

Co-authored-by: Felix Arntz <felixarntz@users.noreply.github.com>

pbearne committed:
Update modules/images/dominant-color/load.php

Co-authored-by: Felix Arntz <felixarntz@users.noreply.github.com>

pbearne committed:
Update modules/images/dominant-color/load.php

Co-authored-by: Felix Arntz <felixarntz@users.noreply.github.com>

pbearne committed:
Update modules/images/dominant-color/load.php

Co-authored-by: Felix Arntz <felixarntz@users.noreply.github.com>

pbearne committed:
ddddd

pbearne committed:
added additional colors to test

pbearne committed:
added additional color to test

pbearne committed:
added additional color to test
